server/filters/get-tasl-markup.js

```diff
@@ -1,10 +1,43 @@
 import getLicenseInfo from './get-license-info';
 
-export default function({title, author, source, license}) {
-  const titleMarkup = source ? `<a rel="cc:attributionURL" property="dc:title" href="${source}">${title || source}</a>. ` : `<span property="dc:title">${title}.</span> `;
-  const authorMarkup = author ? `Credit: <span property="cc:attributionName">${author}.</span> ` : '';
+function setTitle(title, author, sourceLink) {
+  if (!title) return '';
+
+  const byAuthor = author ? `, by ${author}` : '';
+
+  if (sourceLink) {
+    return `<a href="${sourceLink}" property="dc:title" rel="cc:attributionURL">${title}${byAuthor}</a>. `;
+  } else {
+    return `<span property="dc:title">${title}${byAuthor}.</span> `;
+  }
+}
+
+function setSource(sourceName, sourceLink) {
+  if (!sourceName) return '';
+
+  if (sourceLink) {
+    return `Source: <a href="${sourceLink}" rel="cc:attributionURL">${sourceName}</a>. `;
+  } else {
+    return `Source: ${sourceName}. `;
+  }
+}
+
+function setCopyright(copyrightHolder, copyrightLink) {
```

Comment:
Using `set` as the verb here is weird - would something more like `getSourceHtml` make more sense?

Reply:
I knew something was making me feel weird about this – thanks, have updated.